hash.c

```diff
@@ -5097,19 +5097,65 @@ ruby_unsetenv(const char *name)
 /*
  * call-seq:
  *   ENV[name] = value
- *   ENV.store(name, value) -> value
- *
- * Sets the environment variable +name+ to +value+.  If the value given is
- * +nil+ the environment variable is deleted.
- * +name+ must be a string.
  *
+ * Creates, updates, or deletes the named environment variable, returning the value.
+ * Both +name+ and +value+ may be instances of String.
+ * See {Valid Names and Values}[#class-ENV-label-Valid+Names+and+Values].
+ *
+ * - If the named environment variable does not exist:
+ *   - If +value+ is not +nil+, creates the environment variable with +name+ and +value+:
+ *       ENV.clear
+ *       ENV['foo'] ='0' # => '0'
+ *   - If +value+ is +nil+, does nothing.
+ *       ENV.clear
+ *       ENV['foo'] = nil # => nil
+ * - If the named environment variable exists:
+ *   - If +value+ is not +nil+, updates the envirionment variable with value +value+:
+ *       ENV['foo'] = # => '1'
+ *   - If +value+ is +nil+, deletes the environment variable:
+ *       ENV['foo'] = nil # => nil
+ *
+ * Raises an exception if +name+ or +value+ is invalid.
+ * See {Invalid Names and Values}[#class-ENV-label-Invalid+Names+and+Values].
  */
 static VALUE
 env_aset_m(VALUE obj, VALUE nm, VALUE val)
 {
     return env_aset(nm, val);
 }
 
+/*
+ * call-seq:
+ *   ENV.store(name, value) -> value
+ *
+ * ENV.store is an alias for ENV.[]=.
+ *
+ * Creates, updates, or deletes the named environment variable, returning the value.
+ * Both +name+ and +value+ may be instances of String.
+ * See details at {Valid Names and Values}[#class-ENV-label-Valid+Names+and+Values].
+ *
+ * - If the named environment variable does not exist:
+ *   - If +value+ is not +nil+, creates the environment variable with +name+ and +value+:
+ *       ENV.clear
+ *       ENV.store('foo', '0') # => '0'
+ *   - If +value+ is +nil+, does nothing.
+ *       ENV.clear
+ *       ENV.store('foo', nil) # => nil
+ * - If the named environment variable exists:
+ *   - If +value+ is not +nil+, updates the envirionment variable with value +value+:
+ *       ENV.store('foo', '1') # => '1'
+ *   - If +value+ is +nil+, deletes the environment variable:
+ *       ENV.store('foo', nil) # => nil
+ *
+ * Raises an exception if +name+ or +value+ is invalid.
+ * See {Invalid Names and Values}[#class-ENV-label-Invalid+Names+and+Values].
+ */
+static VALUE
+env_store(VALUE obj, VALUE nm, VALUE val)
+{
+    return env_aset(nm, val);
+}
```

Comment:
I've added function env_store here so that it can be documented separately from env_aset_m.  Otherwise, the shared doc would have to have examples for both.

Reply:
I think there is a way to comment methods without implementing a new function, but I don't know it off the top of my head.

Reply:
I've carefully studied the aliases in Hash (5 pairs) and Array (6 pairs), and did not find any example equivalent to what I was after:  an alias pair, with separate call-seq, separate text, and separate examples.

What should we do with this?

Reply:
@tenderlove, Can we get this resolved, one way or another.  I'd like to continue, but I don't want to 'stack' PRs.